### PR TITLE
Update build info

### PR DESCRIPTION
**Pull request by sithlord48**
- Add a page about the build options [preview](https://github.com/sithlord48/wiki-prs/blob/update_BuildInfo/wiki/Build-Options.md)
 - Update Mac os build guide [preview](https://github.com/sithlord48/wiki-prs/blob/master/wiki/Building-on-macOS.md)
 - Update Linux Build Guide [preview](https://github.com/sithlord48/wiki-prs/blob/update_BuildInfo/wiki/Building-on-Linux.md)
 -  Update Windows Build Guide [preview](https://github.com/sithlord48/wiki-prs/blob/update_BuildInfo/wiki/Building-on-Windows.md)

Matching code changes https://github.com/input-leap/input-leap/pull/1981

### CODE REVIEW

File: wiki/Building-on-Linux.md [hunk 1]
```diff
@@ -7,27 +7,32 @@
...
+## Example packages commands
+these can be use to install dependencies on various distros. They may not include what everything for the configuration you are trying to build so DO NOT blindly run these on your machine.
 ### Ubuntu/Debian/Raspbian
 
 ```shell
 sudo apt update && sudo apt upgrade
 sudo apt install git cmake make xorg-dev g++ libcurl4-openssl-dev \
                  libavahi-compat-libdnssd-dev libssl-dev libx11-dev \
-                 qttools5-dev qtbase5-dev libice-dev libsm-dev
+                 qttools6-dev qtbase6-dev libice-dev libsm-dev
```
Comment (AlanMuha) [inline wiki/Building-on-Linux.md]: Regarding the Ubuntu/Debian/Raspbian equivalent packages for `qttools5-dev` and `qtbase5-dev`, seem as if the actual equivalent (at least when I was trying to install from source in my Raspbian system) is `qt6-tools-dev` and `qt6-base-dev`

Comment (sithlord48) [reply wiki/Building-on-Linux.md]: Thank you for the input , updated.

File: wiki/Building-on-Linux.md [hunk 2]
```diff
@@ -7,27 +7,32 @@
...
-(Since Fedora 35, there have been package renames. We leave this to the user, as
-versions of Fedora <= F35 are now EOL, and we're not obligated to support EOL
-versions of Fedora. The following instructions are known to work on F37.)
-
 ```shell
 sudo dnf install git cmake make gcc-c++ xorg-x11-server-devel \
                  libcurl-devel avahi-compat-libdns_sd-devel \
-                 libXtst-devel qt5-qtbase qt5-qtbase-devel  \
-                 qt5-qttools-devel libICE-devel libSM-devel \
+                 libXtst-devel qt6-qtbase qt6-qtbase-devel  \
```
Comment (KoljaFrahm) [inline wiki/Building-on-Linux.md]: Doesn't need qt6-qtbase
libei-devel and libportal-devel should be added

File: wiki/Building-on-macOS.md [hunk 3]
```diff
@@ -4,76 +4,59 @@
...
 
 ### Install dependencies with Homebrew
 
-  - Install packages for Qt5, OpenSSL, and pkgconfig
   ```
-  brew install qt5 openssl pkg-config cmake
+  brew install qt cmake openssl git pkg-config
   ```
 
-### Clone the repository
```
Comment (KoljaFrahm) [inline wiki/Building-on-macOS.md:36]: Why delete this part?

File: wiki/Building-on-Linux.md [hunk 4]
```diff
@@ -40,26 +45,38 @@ sudo dnf groupinstall "Development Tools"
...
 
-### openSUSE
+### openSUSE## Building the Project
 
 ```shell
 sudo zypper install --type pattern devel_basis
 sudo zypper install libdrm-devel libglvnd-devel libICE-devel \
-        libQt5Core-devel libQt5Gui-devel libQt5Network-devel \
-        libqt5-qtbase-common-devel libQt5Widgets-devel libSM-devel \
+        libQt6Core-devel libQt6Gui-devel libQt6Network-devel \
```
Comment (KoljaFrahm) [inline wiki/Building-on-Linux.md]: Are your sure these are the right packages? I don't think they exist

File: wiki/Build-Options.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+Build Options:
+         Option          |            Description                         |   Default Value    | Addtional Requirments |
+:-----------------------------:|:----------------------------------------------:|:------------------:|:---------------------:|
+CMAKE_BUILD_TYPE               | Type of Build that is produced                 | Release            | |
+INPUTLEAP_BUILD_GUI            | Build the Gui                                  | ON                 | Qt|
+QT_DEFAULT_MAJOR_VERSION       | Version of Qt to use for gui  (5 or 6)         | 6                  | Must Build Gui For any Effect |
+INPUTLEAP_BUILD_TESTS          | Builds inputleap tests                         | ON                 | |
+INPUTLEAP_BUILD_EXTERNAL_GTEST | Use System Google Test for test building       | OFF                | Google Tests on your system |
+INPUTLEAP_BUILD_X11            | XWindows support (required) (linux only        | ON                 | xorg-devel |
```
Comment (KoljaFrahm) [inline wiki/Build-Options.md]: ) missing

File: wiki/Building-on-Linux.md [hunk 6]
```diff
@@ -40,26 +45,38 @@ sudo dnf groupinstall "Development Tools"
 sudo dnf install 'dnf-command(config-manager)'
 sudo dnf config-manager --set-enabled PowerTools
 sudo dnf install cmake3 avahi-compat-libdns_sd-devel libX11-devel libXtst-devel \
-                qt5-qtbase-devel libcurl-devel desktop-file-utils openssl-devel
+                qt6-qtbase-devel libcurl-devel desktop-file-utils openssl-devel
 ```
 
-### openSUSE
+### openSUSE## Building the Project
```
Comment (KoljaFrahm) [inline wiki/Building-on-Linux.md]: why does it have building the project here?

File: wiki/Building-on-Linux.md [hunk 7]
```diff
@@ -40,26 +45,38 @@ sudo dnf groupinstall "Development Tools"
...
-        libqt5-qtbase-common-devel libQt5Widgets-devel libSM-devel \
+        libQt6Core-devel libQt6Gui-devel libQt6Network-devel \
+        libqt6-qtbase-common-devel libQt6Widgets-devel libSM-devel \
         libXinerama-devel libXrandr-devel libXrender-devel
 ```
 
 ## Building from source
 
-```shell
-git clone https://github.com/input-leap/input-leap.git
```
Comment (KoljaFrahm) [inline wiki/Building-on-Linux.md:59]: why remove this?

File: wiki/Building-on-Linux.md (hunk 4 above)
Comment (KoljaFrahm) [reply wiki/Building-on-Linux.md]: pkgconfig(Qt6Core)
pkgconfig(Qt6Gui)
pkgconfig(Qt6Network)
pkgconfig(Qt6Platform)
pkgconfig(Qt6Widgets)

I think it also needs pkgconfig(Qt6Linguist)

Comment (KoljaFrahm) [reply wiki/Building-on-Linux.md]: How about replacing most by pkgconfig?

**Review by AlanMuha — COMMENTED** [review]
Maybe specify on the Linux Build that Qt6 is the default option, in the dependency list. Also in Ubuntu section, it still has qt5 dependencies, not qt6s (`qt6-tools-dev`, `qt6-base-dev`)

**Comment by sithlord48** [comment]
> Maybe specify on the Linux Build that Qt6 is the default option, in the dependency list. Also in Ubuntu section, it still has qt5 dependencies, not qt6s (`qt6-tools-dev`, `qt6-base-dev`)

Added the note that Qt6 is default and  removed the ubuntu package install line for now as no release has the libei or libportal without building them.

File: wiki/Building-on-Linux.md [hunk 8]
```diff
@@ -7,59 +15,65 @@
 **Note**: these aren't the only things built at compile time, but merely what the end-user is expected to use.
 
 ## Preparing your system
+This project depends upon the following Make sure you install these before you build. If you get lost you can always check the [github workflow](https://github.com/input-leap/input-leap/blob/master/.github/workflows/builds.yml) to see how we build for your distro. Depending on your distrobution you may need to install `dev` or `devel` pacakges to get the files need to build aginst the dependencies.
```
Comment (KoljaFrahm) [inline wiki/Building-on-Linux.md]: remove "This project depends upon the following"
pacakges
aginst

Comment (KoljaFrahm) [reply wiki/Building-on-Linux.md]: Otherwise Linux section lgtm

Comment (sithlord48) [reply wiki/Building-on-Linux.md]: Should be good now thanks

**Review by KoljaFrahm — APPROVED** [review]

**Pull request merged**